src/Parser/DeathKnight/Shared/RuneTracker.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+
+import RESOURCE_TYPES from 'common/RESOURCE_TYPES';
+import SPELLS from 'common/SPELLS';
+import { formatNumber , formatPercentage } from 'common/format';
+import Wrapper from 'common/Wrapper';
+import SpellIcon from 'common/SpellIcon';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import CastEfficiency from 'Parser/Core/Modules/CastEfficiency';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+
+//TODO
+//add cooldown reduction through spells
+
+/*
+ * Runes are tracked as 3 fake spells with 2 charges to similate 3 runes charging at the same time.
+ * aslong as spells always use the rune pair with the shortest cooldown remaining it should match
+ * its in game functionality.
+ */
+class RuneTracker extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+    castEfficiency: CastEfficiency,
+  };
+
+  timeSpentWithRunesOnCooldown = {};
+  resourceType = RESOURCE_TYPES.RUNES.id;
+
+  on_byPlayer_cast(event) {
+  	event.classResources
+      .filter(resource => resource.type === this.resourceType)
+      .forEach(({ cost }) => {
+        //should add a check here to see if amount matches our rune count.
+        const runeCost = cost || 0;
+      	for(let i = 0; i < runeCost; i++){
+      		this.startCooldown();
+      	}
+      });
+  }
+
+  startCooldown(){
+  	const runeId = this.shortestCooldown;
+  	this.spellUsable.beginCooldown(runeId);
+  }
+
+  get shortestCooldown(){
+  	const runeOneCooldown = this.getCooldown(SPELLS.RUNE_1.id) || 0;
+  	const runeTwoCooldown = this.getCooldown(SPELLS.RUNE_2.id) || 0;
+  	const runeThreeCooldown = this.getCooldown(SPELLS.RUNE_3.id) || 0;
+  	if(runeOneCooldown <= runeTwoCooldown && runeOneCooldown <= runeThreeCooldown){
+  		return SPELLS.RUNE_1.id;
+  	}
+  	if(runeTwoCooldown <= runeThreeCooldown){
+  		return SPELLS.RUNE_2.id;
+  	}
+  	return SPELLS.RUNE_3.id;
+  }
+
+  getCooldown(spellId){
+  	if(!this.spellUsable.isOnCooldown(spellId)){
+  		return null;
+  	}
+  	const chargesOnCooldown = 2 - this.spellUsable.chargesAvailable(spellId);
+  	const cooldownRemaining = this.spellUsable.cooldownRemaining(spellId);
+  	return (chargesOnCooldown - 1) * 10000 + cooldownRemaining;
```

Comment:
Since you're using the static 10s CD, won't this incorrectly scale with haste? Maybe look up the cooldown using the Abilities class.